docs/xdebug.md

```diff
@@ -0,0 +1,85 @@
+    
+    Your settings should look like this. Hit APPLY and OK.
+    ![Debug](../imgs/xdebug/debug-settings.png)    
+
+3. Go to `Preferences > Languages & Frameworks > PHP > Servers`. We will create a new server here. Use these settings:
+    - Name: `docker-debug-server`
+    - Host: `localhost`
+    - Port: `8001`
+    - Use path mappings: yes, select the checkbox
+    - Under project files, select the top-level `archipelago-deployment` directory in the `File/Directory` column.
+    - In the `Absolute path on the server` add `/var/www/html`
+    
+    Hit APPLY and OK and close the window.
+  
+    ![Debug](../imgs/xdebug/server-settings-2.png)    
+ 
+ 4. Go to `Run > Edit Configurations`. Hit the `+` Button to create a new PHP Remote Debug. Name whatever you want, I called mine `docker`. Use these settings:
+    - Filter debug connection by IDE Key: yes, select the checkbox
+    - Server: select `docker-debug-server` from dropdown (we created this in step 3)
+    - IDE Key: `archipelago` (this matches the key set in our container)
+    
+    ![Debug](../imgs/xdebug/edit-configurations.png)    
+ 
+ 5. Validate your connection. With  `Run > Edit Configurations` still open, you can hit the link that says "Validate". Use these settings in the following validation window:
+    - Path to create validation script `<your local path>/archipelago-deployment/web`
+    - Url to validation script: `http://localhost:8001`
+    
+    Hit VALIDATE. You should get a series of green check marks. If you get a warning about missing `php.ini` file, that is OK, our file has a different name in the container (`xdebug.ini`) and is still being read correctly.
+    ![Debug](../imgs/xdebug/validate-2.png)    
+
+ 
+ # Actually Debugging!
+ 1. Hit the button (top right bar of PHPStorm) that looks like a telephone, for `Start Listening for PHP Debug Connections`.
+ 
+      ![Debug](../imgs/xdebug/telephone.png) 
+      
+ 2. Now, you can use `Run > Debug` and select the `docker` named configuration that we created in the previous steps. The debugging console will appear. It will say it is waiting for incoming connection from 'archipelago'
+ .
+       ![Debug](../imgs/xdebug/waiting.png) 
+
+ 3. Go to your Archipelago at `localhost:8001` and refresh the page. If you have breakpoints set, either manually, or from leaving "Break at first line in PHP scripts" checked, you should have output now in the debugger. If you have no breakpoints set, at least the "Waiting for incoming" message will now say "Connected".
+ 
+ 4. If you would like to see the output of your xdebug logs, run the following script:
+ `docker exec -ti esmero-php bash -c 'tail -f /tmp/xdebug.log > /proc/1/fd/2'`
```

Comment:
I feel this is correct. I know i said make the logs part of the container output, but then looking at this it feels better. That way we can separate logs coming from xdebug.log from the general phpfpm ones. But, a catch here. Do we need a cron that clears those logs? Just a question since i have not tested this, just worried the container local storage will end being full of logs after a few days of operation. Any ideas on that?